targets/TARGET_STM/us_ticker_32b.c

```diff
@@ -48,6 +48,10 @@ void us_ticker_set_interrupt(timestamp_t timestamp)
     __HAL_TIM_SET_COMPARE(&TimMasterHandle, TIM_CHANNEL_1, (uint32_t)timestamp);
     // Enable IT
     __HAL_TIM_ENABLE_IT(&TimMasterHandle, TIM_IT_CC1);
+    // Check if timestamp has already passed, and if so, call the IRQ handler immediately
+    if ((int32_t)(timestamp - TIM_MST->CNT) <= 0) {
+        us_ticker_irq_handler();
```

Comment:
@monkiineko 
Hi  - thanks for reporting this point. 
Note that I have just concern that has already been raised with other targets.
us_ticker_set_interrupt might be called already from an interrupt handler, and by calling again the interrupt handler we might end up in a recursive loop ...
Anyway the same limitation applies to current 16 bits timer implementation, so I'm ok to go ahead with it for now and I'll work on an alternative solution. (Basically we need to trig the CC1 interrupt from SW )